src/components/App.js

```diff
@@ -1,31 +1,41 @@
 import React from 'react';
 import { ConnectedRouter } from 'connected-react-router';
-import { Switch } from 'react-router-dom';
+import { Switch, Route } from 'react-router-dom';
 import { Helmet } from 'react-helmet';
 import { ThemeProvider } from 'styled-components';
 
-import { useSession } from 'hooks';
 import history from 'utils/history';
-import RouteFromPath from 'components/routes/RouteFromPath';
-import routes from '../routes';
+import routesPaths from 'constants/routesPaths';
+import HomePage from 'pages/HomePage';
+import LoginPage from 'pages/LoginPage';
+import SignUpPage from 'pages/SignUpPage';
+import NotFoundPage from 'pages/NotFoundPage';
 import theme from '../constants/theme';
 import Header from './common/Header';
+import PrivateRoute from './routes/PrivateRoute';
 
 const App = () => {
-  const { authenticated } = useSession();
-
   return (
     <ThemeProvider theme={theme}>
       <>
         <Helmet>
-          <title>RS React Redux Base</title>
+          <title>Open form webapp</title>
         </Helmet>
         <Header />
         <ConnectedRouter history={history}>
           <Switch>
-            {routes.map((route, index) => (
-              <RouteFromPath key={`route${index}`} {...route} authenticated={authenticated} />
-            ))}
+            <PrivateRoute path={routesPaths.index} exact>
```

Comment:
Routes are defined in [this file](https://github.com/rootstrap/open-form-admin-webapp/blob/master/src/routes.js)

Reply:
But isn't more common to just define the routes as a react component doing like any other stuff in react?
I did it this way in case I wanted to compose the routes in easy way instead,
Like I could do 
`<Route path={routesPaths.signUp}>
      <Header />
          <SignUpPage />
      <Footeer />
</Route>`

Reply:
The routes array is just a convenient way to centralize where you declare your routes. The approach you suggest is also valid, but it can become difficult to find routes when composing them.

Reply:
As I read react router docs, the first implementation they did, you had to declare routes like a static way using an array, and they weren't happy about that, so they did some mayor changes, so you could be able to do this stuff, after all, on the code, you were mapping the routes to end like a component hierarchy, so It wouldn't be more straight forward this way?
And sometimes nesting routes, could be really helpful

Reply:
I think it's best to have structure, the array let's you do everything just fine. The example you said above it's not a really good practice, I get it can be quick and easy but we try to keep everything in components so it's cleaner and more extensible.

Reply:
I agree with coni here. While sometimes nesting routes could come in handy. It also makes flows way less clear and error prone (since you have more than one place that is impacted by the current path).

Reply:
This way will let you do everything just fine too, also, can you explain me why is not a good practice? It still being components, so it will be extensible, and if I gets too bloated, I can extract to new components to have a clear view

Reply:
The best example I can think of is having a nested route that depends on your path, changing that said path and not updating the route match. You would get a really nice bug there. 

Also, There's no case that you could solve by nesting routes that you could not solve using normal components. So I'm not seeing the benefit of breaking the pattern just for the sake of it.

Reply:
I agree with the guys here. If both ways work, and one makes it easier to know where all the routes are, I would go with that one. Besides, we are not straying from the way we already work in other projects.

Reply:
ok guys, I will try to stick to the examples provided on the react redux base